surf/apps/materials/views.py

```diff
@@ -297,6 +298,30 @@ def post(self, request, *args, **kwargs):
         return Response(material_object.applaud_count)
 
 
+class CollectionMaterialPromotionAPIView(APIView):
+    def post(self, request, *args, **kwargs):
+        # only active and authorized users can promote materials in the collection
+        collection_instance = Collection.objects.get(id=kwargs['collection_id'])
+
+        # check whether the material is actually in the collection
+        external_id = kwargs['external_id']
+        collection_materials = CollectionMaterial.objects.filter(
+            collection=collection_instance, material__external_id=external_id)
+        if not collection_materials:
+            raise Http404(f"Collection {collection_instance} does not contain a material with "
+                          f"external id {external_id}, cannot promote or demote")
+
+        # The material should only be in the collection once
+        assert len(collection_materials) == 1, f"Material with id {external_id} is in collection " \
+                                               f"{collection_instance} multiple times."
+        collection_material = collection_materials[0]
+        # promote or demote the material
+        collection_material.featured = not collection_material.featured
+        collection_material.save()
+
+        return Response(serializers.serialize('python', [collection_material]))
```

Comment:
"python" is actually not a serialization format in Django. Defaults to "json" I presume? See: https://docs.djangoproject.com/en/2.2/topics/serialization/#serialization-formats